### PR TITLE
[release/6.0] Suppress clang-16 warnings (backport #81573)

### PR DESCRIPTION
**Pull request by ayakael**
Backports https://github.com/dotnet/runtime/pull/81573 and https://github.com/dotnet/runtime/pull/82461

## Customer impact
Without this, .NET 6 fails to build on platforms that use clang-16

## Testing
Tested on CI and manually on Alpine Linux edge.

## Risk
Low. The flag that is set makes the compiler less strict.

@am11 @omajid ptal.

### CODE REVIEW

**Comment by am11** [comment]
Backport https://github.com/dotnet/runtime/pull/82461 at the same time?

**Comment by carlossanlop** [comment]
cc @tommcdon (see https://github.com/dotnet/runtime/pull/84444#issuecomment-1502090993)

**Review by hoyosjs — APPROVED** [review]

**Review by hoyosjs — APPROVED** [review]

**Comment by am11** [comment]
Failures look related.

File: src/native/corehost/apphost/static/singlefilehost_OSXexports.src [hunk 1]
```diff
@@ -1,11 +0,0 @@
-; Licensed to the .NET Foundation under one or more agreements.
```
Comment (hoyosjs) [inline src/native/corehost/apphost/static/singlefilehost_OSXexports.src:1]: Woops. This file should not be deleted. Rather renamed to src/native/corehost/apphost/static/singlefilehost_freebsdexports.src and you'd need to add the other two exports that got deleted in the next file.

**Review by hoyosjs — CHANGES_REQUESTED** [review]

File: src/native/corehost/apphost/static/singlefilehost_OSXexports.src (hunk 1 above)
Comment (ayakael) [reply src/native/corehost/apphost/static/singlefilehost_OSXexports.src]: Woops indeed! Thanks for catching that, fixed :)

**Review by akoeplinger — APPROVED** [review]

**Comment by akoeplinger** [comment]
Failures look unrelated to me now.

**Comment by carlossanlop** [comment]
Today is code complete for the June Release. @hoyosjs @akoeplinger is this ready to merge? @hoyosjs you have a blocking review.

**Review by hoyosjs — APPROVED** [review]

**Comment by akoeplinger** [comment]
@carlossanlop this is ready 👍

**Pull request merged**